src/coreComponents/common/DataTypes.hpp

```diff
@@ -724,6 +724,38 @@ struct TypeName
   }
 };
 
+
+enum class FieldLocation
```

Comment:
I'm really not in favor of having this here.
This file contains "_technical_" types.
I do not have time right now to further check, but at least I would put this in its own header.

Reply:
well, moving it to its own header is not a big deal. I just don't think it's a good idea to define it in `NeighborCommunicator` as I had tried to do before.

Reply:
Well, I guess we should do different things for `FieldLocation` and `SyncFieldsID`. The `SyncFieldsID` could, indeed, just be defined in the communications module. The location of a dof is a wider concept though and used across the code. The `DofManager` was defining it but if we want to use it in other placess (e.g., the communication routines) we should define it elsewhere coz I don't think we want all those things to depend on the `DofManager`.